File: 3-LongestSubstringWithoutRepeatingCharacters/3-LongestSubstringWithoutRepeatingCharacters.java
// Last updated: 4/3/2026, 2:04:47 PM
class Solution {
    public int lengthOfLongestSubstring(String s) {
        String s1="";
        int maxlen=0, pos=0;
        for(int i=0;i<s.length();i++) {
            if(s1.indexOf(s.charAt(i))==-1) {
                s1=s1+s.charAt(i);
            }
            else {
                if(s1.length()>maxlen) {
                    maxlen=s1.length();
                }
                s1=""+s.charAt(++pos);
                i=pos;
            }
        }
        if(s1.length()>maxlen)
            maxlen=s1.length();
        return maxlen;
    }
}

File: 3-LongestSubstringWithoutRepeatingCharacters/3-LongestSubstringWithoutRepeatingCharacters.cpp
// Last updated: 4/3/2026, 2:04:44 PM
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        
        
        vector<int> characters(256, -1);
        int left = 0, right = 0;
        int n = s.length();
        int maxLen = 0;

        while (right < n) {
            if (characters[s[right]] != -1) {
                left = max(left, characters[s[right]] + 1);
            }
            characters[s[right]] = right;
            maxLen = max(maxLen, right - left + 1);
            right++;
        }
        return maxLen;
    }
};